vllm/worker/tpu_model_runner.py

```diff
@@ -259,19 +312,25 @@ def _dummy_run(
         # in the first run, but can be skipped afterwards as we cache the XLA
         # graphs in the disk (VLLM_XLA_CACHE_PATH).
         if exec_mode.is_prefill():
-            # Prefll
-            torch._dynamo.mark_dynamic(token_ids, 1)
-            torch._dynamo.mark_dynamic(position_ids, 1)
+            # Prefill
+            if self.lora_config is not None:
+                torch._dynamo.config.capture_dynamic_output_shape_ops = True
+            else:
+                torch._dynamo.mark_dynamic(token_ids, 1)
+                torch._dynamo.mark_dynamic(position_ids, 1)
             torch._dynamo.mark_dynamic(attn_metadata.slot_mapping, 1)
         else:
             # Decode
-            torch._dynamo.mark_dynamic(token_ids, 0)
-            torch._dynamo.mark_dynamic(position_ids, 0)
-            torch._dynamo.mark_dynamic(input_lens, 0)
+            if self.lora_config is not None:
+                torch._dynamo.config.capture_dynamic_output_shape_ops = True
```

Comment:
main branch has changed significantly with the addition of a unified kernel, there's no prefill/decode branch anymore. If you find the time, try to rebase your work there.

Reply:
Ah this code path needs to be removed, it's part of v0.